### PR TITLE
Upgrade to Bootstrap 5

### LINKED ISSUE #2845: Upgrade to Bootstrap v5
https://v5.getbootstrap.com/

https://github.com/twbs/bootstrap/projects

Wouldn't be a bad idea to get a head start.

### PR DESCRIPTION
**Pull request by Andrew-Chen-Wang**
## Description

Add bootstrap 5 support, and drop IE and jQuery support (since Bootstrap 5 dropped ie)

Checklist:

- [ ] I've made sure that `tests/test_cookiecutter_generation.py` is updated accordingly (especially if adding or updating a template option)
- [ ] I've updated the documentation or confirm that my change doesn't require any updates

## Rationale

Fixes #2845

### CODE REVIEW

File: {{cookiecutter.project_slug}}/gulpfile.js [hunk 1]
```diff
@@ -32,7 +32,6 @@ function pathsConfig(appName) {
     {%- if cookiecutter.custom_bootstrap_compilation == 'y' %}
     bootstrapSass: `${vendorsRoot}/bootstrap/scss`,
     vendorsJs: [
-      `${vendorsRoot}/jquery/dist/jquery.slim.js`,
       `${vendorsRoot}/popper.js/dist/umd/popper.js`,
```
Comment (steveputman) [inline {{cookiecutter.project_slug}}/gulpfile.js:35]:
```suggestion
      `${vendorsRoot}/@popperjs/core/dist/umd/popper.js`,
```

File: {{cookiecutter.project_slug}}/package.json [hunk 2]
```diff
@@ -5,10 +5,9 @@
   "devDependencies": {
     {% if cookiecutter.js_task_runner == 'Gulp' -%}
     {% if cookiecutter.custom_bootstrap_compilation == 'y' -%}
-    "bootstrap": "4.3.1",
+    "bootstrap": "5.0.2",
     "gulp-concat": "^2.6.1",
-    "jquery": "3.3.1",
-    "popper.js": "1.14.3",
+    "popper.js": "2.9.2",
```
Comment (steveputman) [inline {{cookiecutter.project_slug}}/package.json:10]:
```suggestion
    "@popperjs/core": "2.9.2",
```

**Review by steveputman — COMMENTED** [review]
Node couldn't find Popper 2 as written.

**Comment by Andrew-Chen-Wang** [comment]
thanks @steveputman !

**Comment by Andrew-Chen-Wang** [comment]
@browniebroke would love to see this get merged (ref https://github.com/cookiecutter/cookiecutter-django/issues/2845#issuecomment-966336810)! I think starter templates would benefit from starting off with no jQuery (unlike age old projects which are migrating from jQuery)

**Comment by browniebroke** [comment]
Yes, me too. Last time I tried, I found a couple of issues, but I sadly had less time to look after the project recently... I'll try it again.

**Review by browniebroke — APPROVED** [review]
Fixed a few minor styling issue, LGTM

**Comment by browniebroke** [comment]
Thanks @Andrew-Chen-Wang! 🎉

**Pull request merged**